### PR TITLE
Use SwiftKuery 3

### PR DESCRIPTION
**Pull request by kilnerm**
This PR contains the required changes to use SwiftKuery 3.0

### CODE REVIEW

File: Sources/SwiftKueryORM/Database.swift [hunk 1]
```diff
@@ -65,16 +65,23 @@ public class Database {
...
+    public func getConnection(task: @escaping (Connection?, QueryError?) -> ()) {
         switch connectionStrategy {
-        case .pool(let pool): return pool.getConnection()
-        case .generator(let generator): return generator()
+        case .pool(let pool): return pool.getConnection(poolTask: task)
+        case .generator(let generator): return generator(task)
+        }
+    }
+
+    public func executeTask(task: @escaping (Connection?, QueryError?) -> ()) {
```
Comment (djones6) [inline Sources/SwiftKueryORM/Database.swift]: This needs documenting

File: Sources/SwiftKueryORM/Database.swift [hunk 2]
```diff
@@ -65,16 +65,23 @@ public class Database {
...
     /// Constructor for a custom connection generator
-    public init(generator: @escaping () -> Connection?) {
+    public init(generator: @escaping ((Connection?, QueryError?) -> ()) -> ()) {
         self.connectionStrategy = .generator(generator)
     }
 
     /// Connection getter: either new connection from pool
     /// or call the custom generator
-    public func getConnection() -> Connection? {
+    public func getConnection(task: @escaping (Connection?, QueryError?) -> ()) {
```
Comment (djones6) [inline Sources/SwiftKueryORM/Database.swift]: We should update the documentation comments for `init()` and `getConnection()` to describe the parameters and give usage examples.

File: Sources/SwiftKueryORM/Model.swift [hunk 3]
```diff
@@ -21,1126 +21,1071 @@ import Dispatch
...
-  static func createTable(using db: Database? = nil, _ onCompletion: @escaping (Bool?, RequestError?) -> Void) {
-    var connection: Connection
-    var table: Table
-    do {
-      connection = try Self.getConnection(using: db)
-      table = try Self.getTable()
-    } catch let error {
-      onCompletion(nil, Self.convertError(error))
-      return
+    /// Defaults to the name of the model + "s"
```
Comment (djones6) [inline Sources/SwiftKueryORM/Model.swift]: Because of the indentation change, it's really hard to tell what's actually changed here... is it possible to do it as a separate PR?

My concern is that this is a `public extension` which - I _think_ - means that all its members are by default public.  Any changes here would therefore be to the public API.

**Comment by learning2makethings** [comment]
Any time frame for this? Unfortunately it's beyond my skills to make these changes myself and I think this pull request will fix the dependency hell and inability to build while using the latest version of Swift-Kuery-PostgreSQL which uses SwiftKuery 3 also. Thanks.

**Comment by kilnerm** [comment]
@learning2makethings I split the whitespace changes out prior to the holiday period, first morning back today and I have pulled the other changes into a new branch. Will be adding the necessary documentation shortly and hopefully be in a position get a new tag out this week.
